### PR TITLE
 Limited glob support for enumerators

### PR DESCRIPTION
**Pull request by moadibfr**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | yes
| 🔗 Related issues | #316 
| ❓ Documentation  | yes 

## Description
 Limited glob support.
 Support go glob syntax + ** syntax on file enumerator
 Support go glob syntax on s3 enumerator

### CODE REVIEW

File: pkg/iac/terraform/state/enumerator/file_test.go [hunk 1]
```diff
@@ -20,18 +20,58 @@ func TestFileEnumerator_Enumerate(t *testing.T) {
...
 				"testdata/states/s3/terraform.tfstate",
 				"testdata/states/terraform.tfstate",
 			},
 		},
+		{
+			name: "ext glob !",
+			config: config.SupplierConfig{
+				Path: "testdata/states/**/*.!(backup)",
+			},
+			err: "Unsupported syntax !()",
```
Comment (sundowndev) [inline pkg/iac/terraform/state/enumerator/file_test.go:66]: What are the parentheses about at the end?

File: pkg/iac/terraform/state/enumerator/file_test.go [hunk 2]
```diff
@@ -20,18 +20,58 @@ func TestFileEnumerator_Enumerate(t *testing.T) {
...
 			name: "symlinked folder",
 			config: config.SupplierConfig{
 				Path: "testdata/symlink",
 			},
 			want: []string{
-				"testdata/states/route53/directory/route53.state",
+				"testdata/states/lambda/lambda.tfstate",
 				"testdata/states/s3/terraform.tfstate",
 				"testdata/states/terraform.tfstate",
 			},
```
Comment (sundowndev) [inline pkg/iac/terraform/state/enumerator/file_test.go:77]: Shouldn't the expected value contain `testdata/symlink/` directory ?

**Review by sundowndev — CHANGES_REQUESTED** [review]

File: pkg/iac/terraform/state/enumerator/file_test.go (hunk 1 above)
Comment (moadibfr) [reply pkg/iac/terraform/state/enumerator/file_test.go:66]: maybe it's not clear. It's the syntax that we don't support like !(val1|val2)

File: pkg/iac/terraform/state/enumerator/file_test.go (hunk 2 above)
Comment (moadibfr) [reply pkg/iac/terraform/state/enumerator/file_test.go:77]: no we don't follow symlink I didn't change this behavior

File: pkg/iac/terraform/state/enumerator/file_test.go (hunk 1 above)
Comment (sundowndev) [reply pkg/iac/terraform/state/enumerator/file_test.go:66]: Right, I though it was the returned error

File: pkg/iac/terraform/state/enumerator/file_test.go (hunk 2 above)
Comment (sundowndev) [reply pkg/iac/terraform/state/enumerator/file_test.go:77]: Understood!

**Review by sundowndev — DISMISSED** [review]
LGTM

**Comment by moadibfr** [comment]
user mwarkentin on discord had multiple files on the same s3 bucket were his statefile lays and listing take too much time. Progress bar is being stopped.
I have to send tics to progressbar so it dont stop when listing from s3...

**Comment by eliecharra** [comment]
To rebase on `v0.8` and change target branch

**Review by wbeuil — CHANGES_REQUESTED** [review]
Too bad for the ** syntax on S3. Small typo in one sentence.

**Review by moadibfr — COMMENTED** [review]
LGTM. I wonder if  you could have use just doublestar.glob instead of testing double start presence in the file glob but it's a details.

**Review by eliecharra — DISMISSED** [review]

**Review by eliecharra — APPROVED** [review]

File: pkg/iac/terraform/state/enumerator/file.go [hunk 3]
```diff
@@ -24,58 +23,35 @@ func NewFileEnumerator(config config.SupplierConfig) *FileEnumerator {
...
-// We may use something like this https://pkg.go.dev/github.com/facebookgo/symwalk
-// to follow links, but it allows infinite loop so be careful !
-// If a symlink is given as root path, we will follow it, but symlinks under this path
-// will not be resolved.
 func (s *FileEnumerator) Enumerate() ([]string, error) {
 	path := s.config.Path
+
 	info, err := os.Lstat(path)
-	if err != nil {
+	if isGlob := strings.Contains(path, "*"); !isGlob && err != nil {
```
Comment (moadibfr) [inline pkg/iac/terraform/state/enumerator/file.go:30]: why did you change this and don't use HasMeta?

**Review by eliecharra — APPROVED** [review]
